setup.py

```diff
@@ -62,6 +62,7 @@
         "llvmlite<=0.34.0;python_version>='3.6'",
         "llvmlite<=0.31.0;python_version<'3.6'",
     ],
+    "frozendict": ["frozendict>=1.2"],
```

Comment:
This appears to be a required dependency, not an "extras". Import of the module will fail without this installed.

It appears that its license is compatible with Apache 2.0 https://github.com/slezica/python-frozendict but is there any way we can avoid taking on this extra dependency?

Reply:
I can just initialize this inside of the create_span() method. Or just make a copy inside of these methods. However a copy of default attributes will need to be created with every span. Which shouldn't affect performance too much.

Reply:
I like frozen data structures as a guard against accidentally mutating them, but I agree that adding a dependency may not be worth it.
Note: While you need a copy, you don't need to explicitly call copy(), you could clean up the dict handling into smth. a bit more readable, e.g.:
```
final_attributes = {
  **_default_attributes,
  **client_attributes,
  **job_attributes,
  **attributes,
}
```